pkg/executor/executortype/poolmgr/gp.go

```diff
@@ -60,19 +61,32 @@ type (
 		fsCache                  *fscache.FunctionServiceCache // cache funcSvc's by function, address and podname
 		useSvc                   bool                          // create k8s service for specialized pods
 		useIstio                 bool
-		poolInstanceId           string           // small random string to uniquify pod names
 		runtimeImagePullPolicy   apiv1.PullPolicy // pull policy for generic pool to created env deployment
 		kubernetesClient         *kubernetes.Clientset
 		fissionClient            *crd.FissionClient
-		instanceId               string // poolmgr instance id
 		fetcherConfig            *fetcherConfig.Config
 		stopReadyPodControllerCh chan struct{}
 		readyPodController       cache.Controller
 		readyPodIndexer          cache.Indexer
 		readyPodQueue            workqueue.RateLimitingInterface
+		poolInstanceID           string // small random string to uniquify pod names
+		instanceID               string // poolmgr instance id
+		requestChannel           chan *choosePodRequest
+		stopCh                   context.CancelFunc
+	}
+
+	// serialize the choosing of pods so that choices don't conflict
+	choosePodRequest struct {
+		newLabels       map[string]string
+		responseChannel chan *choosePodResponse
+	}
+	choosePodResponse struct {
+		pod *apiv1.Pod
+		error
 	}
```

Comment:
I think this has been removed in the latest PR which is merged in master now. You might want to rebase and check once @gauravgahlot

Reply:
Yeah right. I have fixed it.